### ISSUE TITLE
ManifoldParameter

### ISSUE THREAD

**Issue by camilobrownpinilla** [Graph-and-Geometric-Learning/hyperbolic-transformer#5]
Hi, 
Great project, and thank you for releasing some nice code! Regarding the difference between manifold parameters and euclidean parameters, I am wondering why manifold parameters are only registered for the HypCLS module? Also, I am probably just not understanding something correctly, but do you mind elaborating on the suggested use case for HypCLS vs. nn.Linear? Thanks!

**Comment by marlin-codes** [comment]
Thank you for your interest! HypCLS defines prototypes as categories and computes negative distances with softmax to obtain probabilities. The prototypes are defined in hyperbolic space and optimized using a Riemannian optimizer. However, for nn.Linear layers, we optimize parameters that exist in Euclidean space, so we employ a Euclidean optimizer. You could certainly define a Euclidean prototype-based classifier and optimize its parameters using a Euclidean optimizer.
